src/Features/Core/Portable/GenerateMember/GenerateParameterizedMember/AbstractGenerateParameterizedMemberService.SignatureInfo.cs

```diff
@@ -124,8 +124,12 @@ public ITypeSymbol DetermineReturnType(CancellationToken cancellationToken)
                 var reservedParameterNames = DetermineParameterNames(cancellationToken)
                                                  .Select(p => p.BestNameForParameter)
                                                  .ToSet(equalityComparer);
-                var newTypeParameterNames = NameGenerator.EnsureUniqueness(
-                    method.TypeParameters.Select(t => t.Name).ToList(), n => !reservedParameterNames.Contains(n));
+
+                using var disposer = ArrayBuilder<string>.GetInstance(out var newTypeParameterNames);
+                NameGenerator.EnsureUniqueness(
+                    method.TypeParameters.SelectAsArray(t => t.Name),
```

Comment:
note: at the *feature* level there sitll may be some allocations.  The main change i made was that the guts of EnsureUniqueness are pooled-allocations.